src/openapi/streaming/connection/transport/websocket-transport.js

```diff
@@ -370,37 +345,40 @@ WebsocketTransport.prototype.start = function(options, callback) {
     this.startedCallback = callback || NOOP;
 
     if (!this.isSupported()) {
-        handleFailure({ message: 'WebSocket Transport is not supported.' });
+        handleFailure.call(this, {
+            message: 'WebSocket Transport is not supported.',
+        });
         return;
     }
 
-    const authorizePromise = this.getAuthorizePromise(this.contextId);
-
-    const url = getWebSocketUrl.call(this);
-
-    log.debug(LOG_AREA, 'Starting transport', { url });
-
     if (this.socket) {
-        // To make sure one socket connection per transport exists, destroy previous one if it exists.
-        destroySocket(this.socket);
-        this.socket = null;
+        log.debug(LOG_AREA, 'only one socket per connection is allowed');
+        return;
     }
 
+    log.debug(LOG_AREA, 'Starting transport');
+
+    const authorizePromise = this.getAuthorizePromise(this.contextId);
+
     authorizePromise.then(() => {
-        this.stateChangedCallback(constants.CONNECTION_STATE_CONNECTING);
-        log.debug(LOG_AREA, 'Creating WebSocket connection', { url });
-        this.socket = new WebSocket(url);
-        configure.call(this);
         this.startedCallback();
+        this.stateChangedCallback(constants.CONNECTION_STATE_CONNECTING);
+        connect.call(this);
     });
 };
 
 WebsocketTransport.prototype.stop = function() {

```

Comment:
Nitpick. My suggestion would be to current stop rename to disconnect and existing disconnect to rename to destroy or close. Reading the code I assumed all disconnect method invocations also caused state changed to disconnected.

Reply:
I am using same interface as in signalR transport.
the main confusion here is due to the fact that we are changing streaming state instead of transport
It has 1:1 maping with singalR